boards/seeedstudio-xiao-esp32c3/Makefile.dep

```diff
@@ -1,9 +1,17 @@
-ifeq (,$(filter stdio_% slipdev_stdio,$(USEMODULE)))
-  USEMODULE += stdio_usb_serial_jtag
+ifneq (,$(filter periph_spi, $(USEMODULE)))
+	ifeq (,$(filter periph_init_buttons, $(DISABLE_MODULE)))
+		DISABLE_MODULE += periph_init_buttons
+
+    MSG="Warning: Using periph_spi on Seeed Studio Xiao ESP32C3 board will disable BUTTON0 due to pin conflict."
+    $(shell $(COLOR_ECHO) "$(COLOR_RED)$(MSG)$(COLOR_RESET)" 1>&2)
+	else ifneq (,$(filter saul_default,$(USEMODULE)))
+		USEMODULE += saul_gpio
+	endif
 endif
```

Comment:
Is it really correct to test for `saul_default` only if `periph_spi` is used? Shouldn't it be tested if `periph_spi` is not used?

```suggestion
ifneq (,$(filter periph_spi, $(USEMODULE)))
	ifeq (,$(filter periph_init_buttons, $(DISABLE_MODULE)))
		DISABLE_MODULE += periph_init_buttons

    MSG="Warning: Using periph_spi on Seeed Studio Xiao ESP32C3 board will disable BUTTON0 due to pin conflict."
    $(info $(COLOR_YELLOW)$(MSG))
	endif
else ifneq (,$(filter saul_default,$(USEMODULE)))
		USEMODULE += saul_gpio
endif
```